src/lsm/lsm_tree.c

```diff
@@ -420,26 +420,32 @@ __wt_lsm_tree_create(WT_SESSION_IMPL *session,
 		WT_ERR_MSG(session, EINVAL,
 		    "LSM merge_min must be less than or equal to merge_max");
 
-	/*
-	 * Set up the config for each chunk.
-	 *
-	 * Make the memory_page_max double the chunk size, so application
-	 * threads don't immediately try to force evict the chunk when the
-	 * worker thread clears the NO_EVICTION flag.
-	 */
-	WT_ERR(__wt_scr_alloc(session, 0, &buf));
-	WT_ERR(__wt_buf_fmt(session, buf,
-	    "%s,key_format=u,value_format=u,memory_page_max=%" PRIu64,
-	    config, 2 * lsm_tree->chunk_max));
-	WT_ERR(__wt_strndup(
-	    session, buf->data, buf->size, &lsm_tree->file_config));
-
-	/* Create the first chunk and flush the metadata. */
-	WT_ERR(__wt_lsm_meta_write(session, lsm_tree));
-
-	/* Discard our partially populated handle. */
-	ret = __lsm_tree_discard(session, lsm_tree, false);
-	lsm_tree = NULL;
+	if (!F_ISSET(S2C(session), WT_CONN_READONLY)) {
+		/*
+		 * Set up the config for each chunk.
+		 *
+		 * Make the memory_page_max double the chunk size, so
+		 * application threads don't immediately try to force evict
+		 * the chunk when the worker thread clears the NO_EVICTION flag.
+		 */
+		WT_ERR(__wt_scr_alloc(session, 0, &buf));
+		WT_ERR(__wt_buf_fmt(session, buf,
+		    "%s,key_format=u,value_format=u,memory_page_max=%" PRIu64,
+		    config, 2 * lsm_tree->chunk_max));
+		WT_ERR(__wt_strndup(
+		    session, buf->data, buf->size, &lsm_tree->file_config));
+
+		/* Create the first chunk and flush the metadata. */
+		WT_ERR(__wt_lsm_meta_write(session, lsm_tree));
+
+		/* Discard our partially populated handle. */
+		ret = __lsm_tree_discard(session, lsm_tree, false);
+		lsm_tree = NULL;
+	} else {
+		F_CLR(lsm_tree, WT_LSM_TREE_MERGES);
```

Comment:
Would it be reasonable to add an assertion to `__wt_lsm_manager_push_entry` that it isn't called in read only mode?